src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.OrderDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Order;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order add(Order order) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot insert Order entity "
+                    + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM Order o "
+                    + "LEFT JOIN FETCH o.tickets "
```

Comment:
Lesh, you need to use distinct here or when you will try to get orders, you are going to have many orders with one ticket, even if those tickets from one order.
And I am not sure but looks like inner joining is enough here.

Reply:
I'm not sure, you're right, I will test.